### DISCUSSION TITLE
Memory released too late for the main cache ?

### DISCUSSION THREAD

**Discussion by cquest** [osm2pgsql-dev/osm2pgsql#1559] [General]
It seems that the memory used by the cache is released late during an osm2pgsql import process.

Once the PG tables are created, I don't think the cache is needed anymore, and the RAM used cannot be used by PG and other caches during the clustering/index creation.

I'm currently finishing a full planet import, and osm2pgsql still uses 65GB of RAM (I allocated 70G for the cache) while line/polygon are creating their index.

**Comment by joto** [comment]
I looked into this and you are right, we could release cache memory earlier. I have no idea how much this will actually improve performance, because the memory is unused and so it should eventually be purged from memory anyway. But it should be fixed in any case. I have some code changes, but they need more tests.

**Comment by doskabouter** [comment]
See also my issue: https://github.com/openstreetmap/osm2pgsql/issues/1532
